### PR TITLE
Fix CVE issues

### PR DESCRIPTION
**Pull request by bachuv**
# Summary
## What changed?
This pull request primarily addresses dependency management and vulnerability mitigation across several test and sample projects. The most significant changes involve updating package versions to patch known security issues, specifically with `System.Drawing.Common` and `jackson-core`, and ensuring compatibility with central package management. Additional updates include bumping `Microsoft.Build` packages to newer versions for consistency and reliability.

**Security and Vulnerability Fixes**

* Downgraded `System.Drawing.Common` to `4.7.3` in multiple `.props` and `.csproj` files to address compatibility and vulnerability concerns.
* Added a custom `Directory.Build.targets` file to forcibly override the vulnerable transitive dependency on `System.Drawing.Common` in auto-generated Azure Functions Worker SDK projects, ensuring version `4.7.3` is used and disabling central package management for the generated project.
* Overrode transitive `jackson-core` dependency in the Java test project to version `2.18.6`, addressing CVE GHSA-72hv-8253-57qq.

**Dependency Management and Compatibility**

* Updated `Microsoft.Build`, `Microsoft.Build.Framework`, and `Microsoft.Build.Tasks.Core` package versions from `17.8.29` to `17.8.43` in test projects for improved consistency and reliability.
* Added an explicit `System.Drawing.Common` reference with `VersionOverride="7.0.0"` in the source generator test project to resolve conflicts with the central pin, since this test project does not ship to customers.

## Why is this change needed?
- Fix CVE issues and vulnerabilities

## Issues / work items
- Resolves #
- Related #

---

# Project checklist
- [x] Documentation changes are not required
  - [ ] Otherwise: Documentation PR is ready to merge and referenced in `pending_docs.md`
- [ ] Release notes are not required for the next release
  - [ ] Otherwise: Notes added to `release_notes.md`
- [x] Backport is not required
  - [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
- [x] All required tests have been added/updated (unit tests, E2E tests)
- [x] No extra work is required to be leveraged by OutOfProc SDKs
  - [ ] Otherwise: Work tracked here: #issue_or_pr_in_each_sdk
- [x] No change to the version of the `WebJobs.Extensions.DurableTask` package
  - [ ] Otherwise: Major/minor updates are reflected in `/src/Worker.Extensions.DurableTask/AssemblyInfo.cs`
- [x] No EventIds were added to `EventSource` logs
  - [ ] Otherwise: Ensure EventIds are within the supported range in the existing Windows infrastructure (validate via deployed telemetry). If needed, extend the range via a PR such as https://msazure.visualstudio.com/One/_git/AAPT-Antares-Websites/pullrequest/7463263?_a=files
- [x] This change should be added to the `v2.x` branch
  - [ ] Otherwise: This change applies exclusively to `WebJobs.Extensions.DurableTask` v3.x and will be retained only in the `dev` and `main` branches
- [ ] Breaking change?
  - [ ] If yes:
    - Impact:
    - Migration guidance:
---

# AI-assisted code disclosure (required)
## Was an AI tool used? (select one)
- [ ] No
- [ ] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [x] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s): VS Code GitHub CoPilot
- AI-assisted areas/files:
- What you changed after AI output:

AI verification (required if AI was used):
- [x] I understand the code and can explain it
- [x] I verified referenced APIs/types exist and are correct
- [x] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [x] I reviewed concurrency/async behavior
- [x] I checked for unintended breaking or behavior changes

---

# Testing
## Automated tests
- Result: Passed / Failed (link logs if failed)

## Manual validation (only if runtime/behavior changed)
- Environment (OS, .NET version, components):
- Steps + observed results:
  1.
  2.
  3.
- Evidence (optional):

---

# Notes for reviewers
- N/A

### CODE REVIEW

File: test/e2e/Apps/BasicDotNetIsolated/Directory.Build.targets [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+    This targets file disables CPM for the generated project and injects a direct PackageReference
+    to override the vulnerable transitive dependency.
+  -->
+  <PropertyGroup Condition="'$(AssemblyName)' == 'Microsoft.Azure.Functions.Worker.Extensions'">
+    <ManagePackageVersionsCentrally>false</ManagePackageVersionsCentrally>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(AssemblyName)' == 'Microsoft.Azure.Functions.Worker.Extensions'">
+    <PackageReference Include="System.Drawing.Common" Version="4.7.2" />
```
[the remark this answers is not in the document]
Comment (bachuv) [reply test/e2e/Apps/BasicDotNetIsolated/Directory.Build.targets]: Fixed.

File: samples/Directory.Packages.props [hunk 2]
```diff
@@ -41,7 +41,7 @@
     <PackageVersion Update="Microsoft.Azure.WebJobs.Script.ExtensionsMetadataGenerator" Version="4.0.1" />
     <PackageVersion Update="Microsoft.Bcl.AsyncInterfaces" Version="10.0.1" />
     <PackageVersion Update="Microsoft.Extensions.Logging.Debug" Version="10.0.0" />
-    <PackageVersion Update="System.Drawing.Common" Version="6.0.0" />
+    <PackageVersion Update="System.Drawing.Common" Version="4.7.3" />
```
[the remark this answers is not in the document]
Comment (bachuv) [reply samples/Directory.Packages.props:44]: Keeping CPM to 4.7.3

File: samples/isolated-unit-tests/IsolatedUnitTest.csproj [hunk 3]
```diff
@@ -9,6 +9,7 @@
   <ItemGroup>
     <FrameworkReference Include="Microsoft.AspNetCore.App" />
     <PackageReference Include="Microsoft.ApplicationInsights.WorkerService" />
+    <PackageReference Include="System.Drawing.Common" VersionOverride="6.0.0" />
```
[the remark this answers is not in the document]
Comment (bachuv) [reply samples/isolated-unit-tests/IsolatedUnitTest.csproj:12]: Keeping CPM to 4.7.3

File: samples/durable-client-managed-identity/aspnetcore-app/ToDoList.csproj [hunk 4]
```diff
@@ -8,6 +8,7 @@
   <!-- To specify a version in the package reference below, simply add `Version = ""`. -->
   <ItemGroup>
     <PackageReference Include="EntityFramework" />
+    <PackageReference Include="System.Drawing.Common" VersionOverride="6.0.0" />
```
[the remark this answers is not in the document]
Comment (bachuv) [reply samples/durable-client-managed-identity/aspnetcore-app/ToDoList.csproj:11]: Keeping CPM to 4.7.3

File: samples/todolist-aspnetcore/ToDoList.csproj [hunk 5]
```diff
@@ -8,6 +8,7 @@
   <!-- To specify a version in the package reference below, simply add `Version = ""`. -->
   <ItemGroup>
     <PackageReference Include="EntityFramework" />
+    <PackageReference Include="System.Drawing.Common" VersionOverride="6.0.0" />
```
[the remark this answers is not in the document]
Comment (bachuv) [reply samples/todolist-aspnetcore/ToDoList.csproj:11]: Keeping CPM to 4.7.3

File: test/e2e/Apps/BasicDotNetIsolated/app.csproj [hunk 6]
```diff
@@ -24,6 +24,7 @@
     <PackageReference Include="Microsoft.Azure.Functions.Worker.ApplicationInsights" />
     <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.DurableTask.SqlServer" />
     <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.DurableTask.AzureManaged" />
+    <PackageReference Include="System.Drawing.Common" VersionOverride="6.0.0" />
```
[the remark this answers is not in the document]
Comment (bachuv) [reply test/e2e/Apps/BasicDotNetIsolated/app.csproj:27]: Keeping CPM to 4.7.3

File: test/Directory.Packages.props [hunk 7]
```diff
@@ -33,7 +33,7 @@
     <PackageVersion Update="Azure.Identity" Version="1.17.1" />
     <PackageVersion Update="Microsoft.Bcl.AsyncInterfaces" Version="10.0.1" />
     <PackageVersion Update="Microsoft.Extensions.Azure" Version="1.10.0" />
-    <PackageVersion Update="System.Drawing.Common" Version="7.0.0" />
+    <PackageVersion Update="System.Drawing.Common" Version="4.7.3" />
     <PackageVersion Update="System.Formats.Asn1" Version="8.0.1" />
     <PackageVersion Update="System.Text.Json" Version="10.0.0" />
   </ItemGroup>
```
[the remark this answers is not in the document]
Comment (bachuv) [reply test/Directory.Packages.props:39]: Fixed.

**Review by sophiatev — APPROVED** [review]
